src/Context/products-context.js

```diff
@@ -0,0 +1,368 @@
+import { useReducer, useContext, createContext } from "react";
+
+const ProductsContext = createContext(null);
```

Comment:
Having null initialization suppresses the purpose can avoid it if you want. it depends on the use-case so do have a look.